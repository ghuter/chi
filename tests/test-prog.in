import libc

main :: fun (argc: i32, argv: []u8) i32 {
	if argc == 2 {
		puts(argv[1])
	}

	else if argc == 2 {
		puts(argv[1])
		puts(argv[2])
	}

	else {
		puts("Hello, world")
	}

	return 0
}
