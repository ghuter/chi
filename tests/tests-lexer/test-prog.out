IMPORT(62)
SPC(8)
IDENTIFIER(5)
<libc>
NEWLINE(6)
NEWLINE(6)
IDENTIFIER(5)
<main>
SPC(8)
COLON(9)
COLON(9)
SPC(8)
FUN(64)
SPC(8)
LPAREN(12)
IDENTIFIER(5)
<argc>
COLON(9)
SPC(8)
IDENTIFIER(5)
<i32>
COMMA(10)
SPC(8)
IDENTIFIER(5)
<argv>
COLON(9)
SPC(8)
LBRACKETS(14)
RBRACKETS(15)
IDENTIFIER(5)
<u8>
RPAREN(13)
SPC(8)
IDENTIFIER(5)
<i32>
SPC(8)
LBRACES(16)
NEWLINE(6)
SPC(8)
IF(55)
SPC(8)
IDENTIFIER(5)
<argc>
SPC(8)
EQUAL(29)
SPC(8)
INT(2)
<2>
SPC(8)
LBRACES(16)
NEWLINE(6)
SPC(8)
IDENTIFIER(5)
<puts>
LPAREN(12)
IDENTIFIER(5)
<argv>
LBRACKETS(14)
INT(2)
<1>
RBRACKETS(15)
RPAREN(13)
NEWLINE(6)
SPC(8)
RBRACES(17)
NEWLINE(6)
NEWLINE(6)
SPC(8)
ELSE(56)
SPC(8)
IF(55)
SPC(8)
IDENTIFIER(5)
<argc>
SPC(8)
EQUAL(29)
SPC(8)
INT(2)
<2>
SPC(8)
LBRACES(16)
NEWLINE(6)
SPC(8)
IDENTIFIER(5)
<puts>
LPAREN(12)
IDENTIFIER(5)
<argv>
LBRACKETS(14)
INT(2)
<1>
RBRACKETS(15)
RPAREN(13)
NEWLINE(6)
SPC(8)
IDENTIFIER(5)
<puts>
LPAREN(12)
IDENTIFIER(5)
<argv>
LBRACKETS(14)
INT(2)
<2>
RBRACKETS(15)
RPAREN(13)
NEWLINE(6)
SPC(8)
RBRACES(17)
NEWLINE(6)
NEWLINE(6)
SPC(8)
ELSE(56)
SPC(8)
LBRACES(16)
NEWLINE(6)
SPC(8)
IDENTIFIER(5)
<puts>
LPAREN(12)
LITERAL(4)
"Hello, world"
RPAREN(13)
NEWLINE(6)
SPC(8)
RBRACES(17)
NEWLINE(6)
NEWLINE(6)
SPC(8)
RETURN(59)
SPC(8)
INT(2)
<0>
NEWLINE(6)
RBRACES(17)
EOI(1)
